extensions/pdh/tests/PerformanceDataMonitorTests.cpp

```diff
@@ -0,0 +1,268 @@
+
+class PerformanceDataMonitorTester {
+ public:
+  PerformanceDataMonitorTester() {
+    LogTestController::getInstance().setTrace<TestPlan>();
+    dir_ = test_controller_.createTempDirectory("/tmp/gt.XXXXXX");
+    plan_ = test_controller_.createPlan();
+    performance_monitor_ = plan_->addProcessor("PerformanceDataMonitor", "pdhsys");
+    putfile_ = plan_->addProcessor("PutFile", "putfile", core::Relationship("success", "description"), true);
+    plan_->setProperty(putfile_, PutFile::Directory.getName(), dir_);
+  }
+
+  void runProcessors() {
+    plan_->runNextProcessor();      // PerformanceMonitor
+    std::this_thread::sleep_for(std::chrono::milliseconds(200));
+    plan_->runCurrentProcessor();   // PerformanceMonitor
+    plan_->runNextProcessor();      // PutFile
+    plan_->runCurrentProcessor();   // PutFile
+  }
+
+  void setPerformanceMonitorProperty(core::Property property, const std::string& value) {
+    plan_->setProperty(performance_monitor_, property.getName(), value);
+  }
+
+  TestController test_controller_;
+  std::string dir_;
+  std::shared_ptr<TestPlan> plan_;
+  std::shared_ptr<core::Processor> performance_monitor_;
+  std::shared_ptr<core::Processor> putfile_;
+};
+
+
+TEST_CASE("PerformanceDataMonitorEmptyPropertiesTest", "[performancedatamonitoremptypropertiestest]") {
+  PerformanceDataMonitorTester tester;
+  tester.runProcessors();
+
+  REQUIRE(tester.test_controller_.getLog().getInstance().contains("No valid counters for PerformanceDataMonitor", std::chrono::seconds(0)));
+
+  uint32_t number_of_flowfiles = 0;
+  auto lambda = [&number_of_flowfiles](const std::string& path, const std::string& filename) -> bool {
+    ++number_of_flowfiles;
+    return true;
+  };
+
+  utils::file::FileUtils::list_dir(tester.dir_, lambda, tester.plan_->getLogger(), false);
+  REQUIRE(number_of_flowfiles == 0);
+}
+
+TEST_CASE("PerformanceDataMonitorPartiallyInvalidGroupPropertyTest", "[performancedatamonitorpartiallyinvalidgrouppropertytest]") {
+  PerformanceDataMonitorTester tester;
+  tester.setPerformanceMonitorProperty(PerformanceDataMonitor::PredefinedGroups, "Disk,CPU,Asd");
+  tester.setPerformanceMonitorProperty(PerformanceDataMonitor::CustomPDHCounters, "\\Invalid\\Counter,\\System\\Processes");
+  tester.runProcessors();
+
+  REQUIRE(tester.test_controller_.getLog().getInstance().contains("Asd is not a valid predefined group", std::chrono::seconds(0)));
+  REQUIRE(tester.test_controller_.getLog().getInstance().contains("Error adding \\Invalid\\Counter to query", std::chrono::seconds(0)));
+
+  uint32_t number_of_flowfiles = 0;
+
+  auto lambda = [&number_of_flowfiles](const std::string& path, const std::string& filename) -> bool {
+    ++number_of_flowfiles;
+    FILE* fp = fopen((path + utils::file::FileUtils::get_separator() + filename).c_str(), "r");
+    REQUIRE(fp != nullptr);
+    char readBuffer[500];
+    rapidjson::FileReadStream is(fp, readBuffer, sizeof(readBuffer));
+    rapidjson::Document d;
```

Comment:
Minor, but I'd avoid the 1 character variable names even in tests for easier search/replace purposes later.

Reply:
yeah thats valid :) changed it to document in [d5959c3](https://github.com/martinzink/nifi-minifi-cpp/commit/d5959c3b1b5d0f377e7b7f2c13cc22a484b2e4a6)